core/src/main/java/org/infinispan/interceptors/distribution/TxDistributionInterceptor.java

```diff
@@ -305,7 +345,7 @@ private Object localGet(InvocationContext ctx, Object key, boolean isWrite,
          }
          if (!ctx.replaceValue(key, ice)) {
             if (isWrite)
-               lockAndWrap(ctx, key, ice, command);
+               entryFactory.wrapEntryForPut(ctx, key, ice, false, command, false);
```

Comment:
Ok, I wasn't sure whether the `LockControlCommand` was created before or after the locking interceptor, but I see now it's right there :)